### DISCUSSION TITLE
Locking down Fields by Workflow State or Limiting Transitions Based on Edited Fields

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16625] [Q&A / Support] [answered]
I am looking to offer a short-cut or "fast path" in my workflow, but this path an only be taken when we are modifying a subset of the model fields (ie these fields have lower scrutiny and can avoid as many steps in an approval process).

Two solutions I came up with:
1. Selectively allow editing of certain fields based on what state of a workflow we are in (ie only the low risk fields are editable)
2. Limit the Transitions offered based on what fields have been edited

How feasible are either of these approaches in Pimcore? Is there some third options that I should consider?

Can (1) be solved using the `objectLayout` on the workflow place?

**Comment by beheist** [comment, marked as answer]
Hey, your first option should be the right way to go, if it fits your requirements. Just set an object layout that allows editing those fields, and you should be fine.

**Reply by cancan101** [reply]
If I use `objectLayout` to limit the fields a user can see / edit, is the user able to just "reload" and choose a different Layout, with more editable fields (assuming their role has that Layout defined)?

**Reply by markus-moser** [reply]
Yes it's always possible to switch between all allowed layouts. If you would like to restrict this in addition to the user permission settings you would need to hook into the pre-send-data event to manipulate the data object detail result json and remove the not allowed layouts there dynamically based on workflow state.
